### PR TITLE
Various Improvements

### PR DESCRIPTION
**Pull request by iamcarbon**
- Configure awaits
- Avoid linq on hot path
- Polyfills string.Contains(char) & StringBuilder.Append(ReadOnlySpan<char>) for .NETSTANDARD2.0
- Simplifies Peek & Consume Lexeme functions
- Allow indexes with patch level differences to be deserialized

### CODE REVIEW

**Comment by iamcarbon** [comment]
Ready for review / feedback.

File: LunrCore/QueryLexer.cs [hunk 1]
```diff
@@ -154,11 +154,11 @@ private string SliceString()
             foreach (int escapeCharposition in _escapeCharPositions)
             {
                 int sliceEnd = escapeCharposition;
-                subSlices.Append(_str.Substring(sliceStart, sliceEnd - sliceStart));
+                subSlices.Append(_str.AsSpan(sliceStart, sliceEnd - sliceStart));
```
Comment (iamcarbon) [inline LunrCore/QueryLexer.cs:157]: Requires a polyfill for .NETSTANDARD2.0

File: LunrCore/LunrCore.csproj [hunk 2]
```diff
@@ -16,6 +16,7 @@
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
     <LangVersion>9</LangVersion>
     <Nullable>enable</Nullable>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (bleroy) [inline LunrCore/LunrCore.csproj:19]: Any way we can do this only for NS2? Or better, don't do the polyfills at all and #ifdef the code instead. I'd really really prefer not to allow unsafe.

File: LunrCore/Polyfills/StringExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+{
+    internal static class StringExtensions
+    {
+        public static bool Contains(this string text, char value)
+        {
+            return text.IndexOf(value) > -1;
+        }
+    }
+}
+#endif
```
Comment (bleroy) [inline LunrCore/Polyfills/StringExtensions.cs:12]: The benefits of this new method seem extremely slim. Can't we just leave the IndexOfs in place? Are there any benefits besides the slight readability improvement?

**Review by bleroy — CHANGES_REQUESTED** [review]
I really like some of those changes, but I'd prefer we avoided unsafe and the polyfills.

**Comment by iamcarbon** [comment]
Agreed on keeping the library safe. The allocation reductions for StringBuilder.Append are unmarkable. 
There's a small perf improvement using Contains on modern runtimes as it skips logic to get the index position. 

I'll do some benchmarking and figure out how many allocations we're saving with the LINQ changes soon -- and open up a new PR with a clean history.

**Pull request closed without merging**